src/validateRegisterForm.test.js

```diff
@@ -26,5 +26,204 @@ describe(`Function 'validateRegisterForm':`, () => {
+    });
+
+    it(`should return error for email with invalid special characters`, () => {
+      const invalidEmail = validateRegisterForm('test@!mail.com', 'P@ssword1!');
+
+      expect(invalidEmail.code).toBe(422);
+      expect(invalidEmail.message).toBe('Email is invalid.');
+    });
+
+    it(`should return success for a valid email with periods`, () => {
+      const isValid = validateRegisterForm('test.mail@mail.com', 'P@ssword1!');
+
+      expect(isValid.code).toBe(200);
+      expect(isValid.message).toBe('Email and password are valid.');
+    });
+
+    it(`should return error for an email with periods at the start`, () => {
+      const invalidEmail = validateRegisterForm('.test@mail.com', 'P@ssword1!');
+
+      expect(invalidEmail.code).toBe(422);
+      expect(invalidEmail.message).toBe('Email is invalid.');
+    });
+
+    it(`should return error for an email with periods at the end`, () => {
+      const invalidEmail = validateRegisterForm('test.@mail.com', 'P@ssword1!');
+
+      expect(invalidEmail.code).toBe(422);
+      expect(invalidEmail.message).toBe('Email is invalid.');
+    });
+
+    it(`should return error for an email with double dots`, () => {
+      const invalidEmail = validateRegisterForm(
+        'test..mail@mail.com', 'P@ssword1!');
+
+      expect(invalidEmail.code).toBe(422);
+      expect(invalidEmail.message).toBe('Email is invalid.');
+    });
+
+    it(`should return success for a valid email with an "@" symbol`, () => {
+      const isValid = validateRegisterForm('test@mail.com', 'P@ssword1!');
+
+      expect(isValid.code).toBe(200);
+      expect(isValid.message).toBe('Email and password are valid.');
+    });
+
+    it(`should return error for an email without an "@" symbol`, () => {
+      const invalidEmail = validateRegisterForm('testmail.com', 'P@ssword1!');
+
+      expect(invalidEmail.code).toBe(422);
+      expect(invalidEmail.message).toBe('Email is invalid.');
+    });
+
+    it(`should return success for a valid email with a top-level'
+      + 'domain not starting with a dot`, () => {
+      const isValid = validateRegisterForm('test@mail.com', 'P@ssword1!');
+
+      expect(isValid.code).toBe(200);
+      expect(isValid.message).toBe('Email and password are valid.');
+    });
+
+    it(`should return success for valid email not starting with a dot`, () => {
+      const isValid = validateRegisterForm('test@mail.com', 'P@ssword1!');
+
+      expect(isValid.code).toBe(200);
+      expect(isValid.message).toBe('Email and password are valid.');
+    });
+
+    it(`should return error for an email starting with a dot`, () => {
+      const invalidEmail = validateRegisterForm('.test@mail.com', 'P@ssword1!');
+
+      expect(invalidEmail.code).toBe(422);
+      expect(invalidEmail.message).toBe('Email is invalid.');
+    });
+
+    it(`should return success for a valid email without double dots`, () => {
+      const isValid = validateRegisterForm('test@mail.com', 'P@ssword1!');
+
```

Comment:
This is the third test about "double dots" in email. I got a little confused. Some of them are definitely not necessary